src/fenicsxconcrete/boundary_conditions/boundary.py

```diff
@@ -86,15 +84,17 @@ def line_at(coordinates : list, dims : list):
     assert dims[0] != dims[1]
 
     def boundary(x):
-        return np.logical_and(np.isclose(x[dims[0]], coordinates[0]),
-                              np.isclose(x[dims[1]], coordinates[1]))
+        return np.logical_and(
+            np.isclose(x[dims[0]], coordinates[0]),
+            np.isclose(x[dims[1]], coordinates[1]),
+        )
 
     return boundary
 
 
 def within_range(
-    start: typing.Union[typing.Iterable[int], typing.Iterable[float]],
-    end: typing.Union[typing.Iterable[int], typing.Iterable[float]],
+    start: typing.Iterable[int] | typing.Iterable[float],
+    end: typing.Iterable[int] | typing.Iterable[float],
```

Comment:
when should we use `list` and when `typing.Iterable`?

Reply:
depends on use case :)